sklearn/neural_network/tests/test_mlp.py

```diff
@@ -437,16 +449,38 @@ def test_predict_proba_multi():
     assert_array_equal(y_log_proba, np.log(y_proba))
 
 
+def test_predict_proba_multilabel():
+    # Test that predict_proba works as expected for multilabel.
+    # Multilabel should not use softmax which makes probabilities sum to 1
+    X, y = make_multilabel_classification(n_samples=50, random_state=0,
+                                          return_indicator=True)
+
+    clf = MLPClassifier(hidden_layer_sizes=5)
+    clf.fit(X, y)
+    y_proba = clf.predict_proba(X)
+    y_log_proba = clf.predict_log_proba(X)
+
+    (n_samples, n_classes) = y.shape[0], y.shape[1]
+
+    proba_max = y_proba.argmax(axis=1)
+    proba_log_max = y_log_proba.argmax(axis=1)
+
+    assert_greater((y_proba.sum(1) - 1).dot(y_proba.sum(1) - 1), 1e-10)
+    assert_equal(y_proba.shape, (n_samples, n_classes))
+    assert_array_equal(proba_max, proba_log_max)
+    assert_array_equal(y_log_proba, np.log(y_proba))
```

Comment:
If you use a larger enough model (to avoid under-fitting) and the l-bfgs solver, then you can actually check that 0.5 proba thresholding works as expected on multilabel data:

```python
>>> clf = MLPClassifier(algorithm='l-bfgs', hidden_layer_sizes=30, random_state=0).fit(X, y)
>>> y_proba = clf.predict_proba(X)
>>> assert_array_equal(y_proba > 0.5, y)
```